### PR TITLE
Rectify the four-byte alignment issue in the fuzzy testing.

### PR DESCRIPTION
**Pull request by Xiaohanjlll**
Fix: #2690

### CODE REVIEW

File: unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c [hunk 1]
```diff
@@ -20,7 +20,15 @@
 size_t libspdm_alignment_size(size_t size)
 {
     size_t alignment;
+    size_t max_buffer_size;
+
     alignment = LIBSPDM_TEST_ALIGNMENT;
+    max_buffer_size = libspdm_get_max_buffer_size();
+
+    /* In the situation where max_buffer_size is not four-byte aligned, reserve sufficient size for the buffer_size */
+    if (size > max_buffer_size - LIBSPDM_TEST_ALIGNMENT && (size & (alignment - 1)) != 0) {
```
Comment (jyao1) [inline unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c:29]: I dont understand why you mix the usage of `alignment` and `LIBSPDM_TEST_ALIGNMENT`.

If you already have alignment, then please always use it.

Comment (jyao1) [inline unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c:29]: please always put () for &&, to make logic clear.

File: unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c [hunk 2]
```diff
@@ -20,7 +20,15 @@
...
+        size -= LIBSPDM_TEST_ALIGNMENT;
```
Comment (jyao1) [inline unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c:30]: use alignment.

File: unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c (hunk 1 above)
Comment (Xiaohanjlll) [reply unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c]: Thanks, I have already added ( )

Comment (Xiaohanjlll) [reply unit_test/fuzzing/spdm_unit_fuzzing_common/toolchain_harness.c]: Thanks, I have already modified it.

**Pull request merged**